### PR TITLE
Action: Definition of Done

### PR DESCRIPTION
**Pull request by ken-mellem**
# Description

This PR introduces a new Action which verifies that the Definition of Done Checklist exists in the Description of the PR, and will fail until all tasks are checked.

Completes: [AB#33717](https://statoil-proview.visualstudio.com/9949ad5e-7d15-4531-901e-296574079ee1/_workitems/edit/33717)

### CODE REVIEW

**Comment by ken-mellem** [comment]
It would be cool if the Description of a Feature and its UAC are copied into the description of a the PR when linked to the appropriate AB# item.... avoid double work.

File: dod.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+dod:
+  - 'PR title and description are to the point and understandable'
+  - 'Link to Azure WorkItem AB#XXXX'
```
Comment (Noggling) [inline dod.yaml:3]: This may be redundant should add bot for this.

Comment (Noggling) [reply dod.yaml:3]: maybe add [ ] I have performed a self-review of my own code.

File: .github/workflows/defDone.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+jobs:
+  check-dod:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Clone Repo
+        uses: actions/checkout@v3
+      - name: Check DoD
+        uses: platisd/definition-of-done@v2.0.0
+        with:
+          dod_yaml: '../dod.yaml'
```
Comment (Noggling) [inline .github/workflows/defDone.yml:15]: remove ../

**Comment by Noggling** [comment]
add this 
# Verify that a valid Azure Board Work Item Type Story exists

name: ab-story-validation

on:
    pull_request:
        types: [opened, edited, reopened]

jobs:
    ab-storyid:
        runs-on: ubuntu-latest
        name: 'Validate Azure Board StoryID'
        steps:
            - name: ab-story-validation
              uses: emagers/ab-story-validation-action@v2.0.1
              with:
                  GITHUB_TOKEN: '${{secrets.GITHUB_TOKEN}}'

**Comment by ken-mellem** [comment]
> add this 
# Verify that a valid Azure Board Work Item Type Story exists

name: ab-story-validation

on:
    pull_request:
        types: [opened, edited, reopened]

jobs:
    ab-storyid:
        runs-on: ubuntu-latest
        name: 'Validate Azure Board StoryID'
        steps:
            - name: ab-story-validation
              uses: emagers/ab-story-validation-action@v2.0.1
              with:
                  GITHUB_TOKEN: '${{secrets.GITHUB_TOKEN}}' 

Will do - on a new PR

**Pull request merged**